### ISSUE TITLE
OtherSource - clrstack

### ISSUE THREAD

**Issue by wmascia** [microsoft/perfview#1010]
We only get native frames with 'k' in the public debugger, so I added a ClrStackSource class (and friends) to handle !clrstack output for customer scenarios.  Can we add this?

                var framePattern = new Regex(@"\b(([^( \r\n)])+)\(.*\)$");
                
                ...

                if (match.Success && newCallStackFound)
                {
                    var methodName = match.Value;
                    var firstNamespaceSegment = methodName.Split('.')[0];
                    var moduleIndex = Interner.ModuleIntern(firstNamespaceSegment);
                    var frameIndex = Interner.FrameIntern(methodName, moduleIndex);

I am by no means a regex expert, so any feedback is welcome.

**Comment by brianrob** [comment]
@wmascia, in general, we're willing to take any file format that benefits from PerfView's view capabilities.  Can you share a bit more about what the scenario looks like?  It sounds like you're getting stacks out of the debugger using !clrstack.  I'm assuming that you're getting many of these and want to aggregate them.

**Comment by wmascia** [comment]
@brianrob correct.  !clrstack is needed instead of k/kc for .NET apps when using public dbg binaries.  Initially I started getting stacks only inscenarios where PerfView etl collection w/ threadtime wasn't working perfectly for some reason (missing frames, broken stacks, not available on down-level OS, etc.).  Now I tend to get it by default to supplement the etl trace.
